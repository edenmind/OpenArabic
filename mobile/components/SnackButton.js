import * as React from 'react'

// eslint-disable-next-line import/namespace
import { StyleSheet, View } from 'react-native'

import PropTypes from 'prop-types'
import { Snackbar } from 'react-native-paper'

const SnackButton = (props) => {
  return (
    <View style={styles.container}>
      <Snackbar visible={props.visible} onDismiss={props.onDismissSnackBar}>
        {props.text}
      </Snackbar>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-between'
  }
})

export default SnackButton

SnackButton.propTypes = {
  visible: PropTypes.bool.isRequired,
  onDismissSnackBar: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired
}
