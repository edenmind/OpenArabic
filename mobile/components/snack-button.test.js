import { describe, it, jest } from '@jest/globals'
import ShallowRenderer from 'react-test-renderer/shallow' // ES6
import React from 'react'
import SnackButton from './snack-button.js'

jest.useFakeTimers()
describe('<SnackButton />', () => {
  it('renders without crashing', () => {
    // @ts-ignore
    const renderer = new ShallowRenderer()
    renderer.render(<SnackButton />)
  })
})
