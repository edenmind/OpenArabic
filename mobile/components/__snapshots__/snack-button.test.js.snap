// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnackButton /> renders correctly 1`] = `
<View>
  <RCTSafeAreaView
    pointerEvents="box-none"
    style={
      [
        {
          "bottom": 0,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "alignSelf": undefined,
          "bottom": undefined,
          "left": undefined,
          "position": undefined,
          "right": undefined,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 6,
          "top": undefined,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 2,
          }
        }
      >
        <View
          accessibilityLiveRegion="polite"
          collapsable={false}
          pointerEvents="box-none"
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgb(202, 200, 165)",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "margin": 8,
              "opacity": 0,
              "transform": [
                {
                  "scale": 0.9,
                },
              ],
            }
          }
          styles={
            {
              "backgroundColor": "rgb(202, 200, 165)",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                [
                  {
                    "flex": 1,
                    "marginLeft": 16,
                    "marginVertical": 14,
                  },
                  {
                    "color": "rgb(28, 28, 23)",
                    "marginRight": 16,
                  },
                ],
              ]
            }
          >
            abc
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
